wp-content/civi-extensions/goonjcustom/Civi/DroppingCenterService.php

```diff
@@ -70,6 +76,114 @@ private static function generateDroppingCenterQrCode($id) {
     self::generateQrCode($data, $id, $saveOptions);
   }
 
+  private static function findOfficeId(array $array) {
+    $filteredItems = array_filter($array, fn($item) => $item['entity_table'] === 'civicrm_eck_collection_source_vehicle_dispatch');
+
+    if (empty($filteredItems)) {
+      return FALSE;
+    }
+
+    $goonjOfficeId = CustomField::get(FALSE)
+      ->addSelect('id')
+      ->addWhere('custom_group_id:name', '=', 'Camp_Vehicle_Dispatch')
+      ->addWhere('name', '=', 'To_which_PU_Center_material_is_being_sent')
+      ->execute()
+      ->first();
+
+    if (!$goonjOfficeId) {
+      return FALSE;
+    }
+
+    $goonjOfficeFieldId = $goonjOfficeId['id'];
+
+    $goonjOfficeIndex = array_search(TRUE, array_map(fn($item) =>
+        $item['entity_table'] === 'civicrm_eck_collection_source_vehicle_dispatch' &&
+        $item['custom_field_id'] == $goonjOfficeFieldId,
+        $filteredItems
+    ));
+
+    return $goonjOfficeIndex !== FALSE ? $filteredItems[$goonjOfficeIndex] : FALSE;
+  }
+
+  public static function mailNotificationToMmt($op, $groupID, $entityID, &$params) {
```

Comment:
I need to update the check for the subtype ID here because it is already implemented in the collection camp. To avoid any conflicts, this function should only run for the dropping center case.